xbmc/addons/kodi-addon-dev-kit/include/kodi/gui/gl/Shader.h

```diff
@@ -0,0 +1,355 @@
+  {
+    GLint params[4];
+
+    Free();
+
+    if (m_source.length()==0)
+      return true;
+
+    m_pixelShader = glCreateShader(GL_FRAGMENT_SHADER);
+
+    GLsizei count = 0;
+    const char *sources[3];
+    if (!extraBegin.empty())
+      sources[count++] = extraBegin.c_str();
+    if (!m_source.empty())
+      sources[count++] = m_source.c_str();
+    if (!extraEnd.empty())
+      sources[count++] = extraEnd.c_str();
+
+    glShaderSource(m_pixelShader, count, sources, 0);
+    glCompileShader(m_pixelShader);
+    glGetShaderiv(m_pixelShader, GL_COMPILE_STATUS, params);
+    if (params[0] != GL_TRUE)
+    {
+      GLchar log[LOG_SIZE];
+      glGetShaderInfoLog(m_pixelShader, LOG_SIZE, nullptr, log);
+      kodi::Log(ADDON_LOG_ERROR, "CPixelShader::%s: %s", __FUNCTION__, log);
+      fprintf(stderr, "CPixelShader::%s: %s\n", __FUNCTION__, log);
+      m_lastLog = log;
+      m_compiled = false;
+    }
+    else
+    {
+      GLchar log[LOG_SIZE];
+      glGetShaderInfoLog(m_pixelShader, LOG_SIZE, nullptr, log);
+      m_lastLog = log;
+      m_compiled = true;
+    }
+    return m_compiled;
+  }
+
+  GLuint Handle() override { return m_pixelShader; }
+
+protected:
+  GLuint m_pixelShader = 0;
+};
+//------------------------------------------------------------------------
+
+//========================================================================
+/// CShaderProgram
+class ATTRIBUTE_HIDDEN CShaderProgram
+{
+public:
+  CShaderProgram() = default;
+  CShaderProgram(const std::string &vert, const std::string &frag)
+  {
+    LoadShaderFiles(vert, frag);
+  }
+
+  virtual ~CShaderProgram()
+  {
+    ShaderFree();
+  }
+
+  bool LoadShaderFiles(const std::string &vert, const std::string &frag)
+  {
+    if (!kodi::vfs::FileExists(vert) || !m_pVP.LoadSource(vert))
+    {
+      kodi::Log(ADDON_LOG_ERROR, "%s: Failed to load '%s'", __func__, vert.c_str());
+      return false;
+    }
+
+    if (!kodi::vfs::FileExists(frag) || !m_pFP.LoadSource(frag))
+    {
+      kodi::Log(ADDON_LOG_ERROR, "%s: Failed to load '%s'", __func__, frag.c_str());
+      return false;
+    }
+
+    return true;
+  }
+
+  bool CompileAndLink(const std::string& vertexExtraBegin = "",
+                      const std::string& vertexExtraEnd = "",
+                      const std::string& fragmentExtraBegin = "",
+                      const std::string& fragmentExtraEnd = "")
+  {
+    GLint params[4];
+
+    // free resources
+    ShaderFree();
+    m_ok = false;
+
+    // compiled vertex shader
+    if (!m_pVP.Compile(vertexExtraBegin, vertexExtraEnd))
+    {
+      kodi::Log(ADDON_LOG_ERROR, "GL: Error compiling vertex shader");
+      return false;
+    }
+
+    // compile pixel shader
+    if (!m_pFP.Compile(fragmentExtraBegin, fragmentExtraEnd))
+    {
+      m_pVP.Free();
+      kodi::Log(ADDON_LOG_ERROR, "GL: Error compiling fragment shader");
+      return false;
+    }
+
+    // create program object
+    if (!(m_shaderProgram = glCreateProgram()))
```

Comment:
I know it's personal style, but I hate assignments in "if" statements. This decreases readability and one has to take a deeper look to understand whether actually `==` was intended.